src/EventGrid/EventGrid/Domain/NewAzureEventGridDomainKey.cs

```diff
@@ -0,0 +1,118 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(
+        VerbsCommon.New,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomainKey",
+        DefaultParameterSetName = DomainNameParameterSet,
+        SupportsShouldProcess = true), 
+    OutputType(typeof(DomainSharedAccessKeys))]
+
+    public class NewAzureEventGridDomainKey : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            ParameterSetName = DomainNameParameterSet,
+            HelpMessage = EventGridConstants.ResourceGroupNameHelp)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            ParameterSetName = DomainNameParameterSet,
+            HelpMessage = EventGridConstants.DomainNameHelp)]
+        [ResourceNameCompleter("Microsoft.EventGrid/domains", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string DomainName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            ParameterSetName = DomainNameParameterSet,
+            HelpMessage = EventGridConstants.KeyNameHelp)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.KeyNameHelp,
+            ParameterSetName = DomainInputObjectParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.KeyNameHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainInputObjectHelp,
+            ParameterSetName = DomainInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSDomain InputObject { get; set; }
```

Comment:
@ahamad-MS I would rename this to be `-DomainObject`

Reply:
I rather keep this as InputObject to be consistent with corresponding topic one unless this is breaking something?

Reply:
Having the parameter named `-InputObject` won't break anything, but it may be confusing to users -- the value for `-InputObject` is usual the representation of the element that the cmdlet is acting upon. For example, a cmdlet called `Remove-AzFoo` would have the user provide the `Foo` object that they want to delete, and a cmdlet called `Update-AzFoo` would have the user provide the `Foo` object that they want to update. We don't advise having an `-InputObject` parameter for `New-*` cmdlets since you wouldn't pass the object representation of the resource you're trying to create (since it would have already been created). In this case specifically, the object that the user would provide is the parent object, which is an acceptable pattern, but we want the name to make this distinction clear. In [this example we created](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/new-cmdlet-example.md#child-resource), you can see we follow a similar pattern by having a `-TopLevelResourceObject` parameter, which would be `-DomainObject` in this case.

Let me know if you have any questions about this pattern.

Reply:
OK. Changed.